lib/runner.js

```diff
@@ -96,7 +96,9 @@ Runner.prototype.run = function () {
 		testCount: 0
 	};
 
-	return Promise.resolve(this.tests.buildPhases(this._addTestResult, this._bail).run()).then(function () {
+	this.tests.on('test', this._addTestResult);
+
+	return Promise.resolve(this.tests.build(this._bail).run()).then(function () {
```

Comment:
`Sequence#run()` and `Concurrent#run()` behavior is on purpose, @jamestalmage says it affects performance to create promises when they are not required.